oss-internship-2020/sapi_lodepng/examples/main_sandboxed.cc

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cassert>
+#include <filesystem>
+#include <iostream>
+
+#include "sandbox.h"
+#include "sandboxed_api/util/flag.h"
+
+ABSL_DECLARE_FLAG(string, sandbox2_danger_danger_permit_all);
+ABSL_DECLARE_FLAG(string, sandbox2_danger_danger_permit_all_and_log);
+ABSL_FLAG(string, images_path, std::filesystem::current_path().string(),
+          "path to the folder containing test images");
+
+void generate_one_step(SapiLodepngSandbox &sandbox, LodepngApi &api) {
+  constexpr unsigned int width = 512, height = 512,
+                         img_len = width * height * 4;
+  std::vector<unsigned char> image(img_len);
+
+  for (int y = 0; y < height; ++y) {
+    for (int x = 0; x < width; ++x) {
+      image[4 * width * y + 4 * x + 0] = 255 * !(x & y);
+      image[4 * width * y + 4 * x + 1] = x ^ y;
+      image[4 * width * y + 4 * x + 2] = x | y;
+      image[4 * width * y + 4 * x + 3] = 255;
+    }
+  }
+
+  // encode the image
+  sapi::v::Array<unsigned char> sapi_image(image.data(), img_len);
+  sapi::v::UInt sapi_width(width), sapi_height(height);
+  std::string filename = "/output/out_generated1.png";
```

Comment:
const std::string or just use "/output/out_generated1.png" in the sapi_filename constructor. There's probably no benefit in having otherwise unused variable here